bin/ch/Helpers.cpp

```diff
@@ -4,6 +4,212 @@
+{
+    size_t dpos = TTDHostStringLength(dst);
+    size_t srcLength = TTDHostStringLength(src);
+    size_t srcByteLength = srcLength * sizeof(TTDHostCharType);
+
+    memcpy_s(dst + dpos, srcByteLength, src, srcByteLength);
+    dst[dpos + srcLength] = _u('\0');
+}
+
+void TTDHostAppendWChar(TTDHostCharType* dst, const wchar* src)
+{
+    size_t dpos = TTDHostStringLength(dst);
+    size_t srcLength = wcslen(src);
+
+    for(size_t i = 0; i < srcLength; ++i)
+    {
+        dst[dpos + i] = (char16)src[i];
+    }
+    dst[dpos + srcLength] = _u('\0');
+}
+
+void TTDHostAppendAscii(TTDHostCharType* dst, const char* src)
+{
+    size_t dpos = TTDHostStringLength(dst);
+    size_t srcLength = strlen(src);
+    for(size_t i = 0; i < srcLength; ++i)
+    {
+        dst[dpos + i] = (char16)src[i];
+    }
+    dst[dpos + srcLength] = _u('\0');
+}
+
+void TTDHostBuildCurrentExeDirectory(TTDHostCharType* path, size_t pathBufferLength)
+{
+    wchar exePath[MAX_PATH];
+    GetModuleFileName(NULL, exePath, MAX_PATH);
+
+    size_t i = wcslen(exePath) - 1;
+    while(exePath[i] != TTDHostPathSeparatorChar)
+    {
+        --i;
+    }
+    exePath[i + 1] = _u('\0');
+
+    TTDHostAppendWChar(path, exePath);
+}
+
+JsTTDStreamHandle TTDHostOpen(const TTDHostCharType* path, bool isWrite)
+{
+    FILE* res = nullptr;
+    _wfopen_s(&res, path, isWrite ? _u("w+b") : _u("r+b"));
+
+    return (JsTTDStreamHandle)res;
+}
+
+#define TTDHostCWD(dst) _wgetcwd(dst, MAX_PATH)
+#define TTDDoPathInit(dst)
+#define TTDHostTok(opath, TTDHostPathSeparator, context) wcstok_s(opath, TTDHostPathSeparator, context)
+#define TTDHostStat(cpath, statVal) _wstat(cpath, statVal)
+
+#define TTDHostMKDir(cpath) _wmkdir(cpath)
+#define TTDHostCHMod(cpath, flags) _wchmod(cpath, flags)
+#define TTDHostRMFile(cpath) _wremove(cpath)
+
+#define TTDHostFindFirst(strPattern, FileInformation) _wfindfirst(strPattern, FileInformation)
+#define TTDHostFindNext(hFile, FileInformation) _wfindnext(hFile, FileInformation)
+#define TTDHostFindClose(hFile) _findclose(hFile)
+
+#define TTDHostDirInfoName(FileInformation) FileInformation.name
+
+#define TTDHostRead(buff, size, handle) fread_s(buff, size, 1, size, (FILE*)handle);
+#define TTDHostWrite(buff, size, handle) fwrite(buff, 1, size, (FILE*)handle)
+#else
+#include <unistd.h>
+#include <cstring>
+#include <libgen.h>
+#include <dirent.h>
+
+#include <sys/stat.h>
+
+typedef utf8char_t TTDHostCharType;
+typedef struct dirent* TTDHostFileInfo;
+typedef DIR* TTDHostFindHandle;
+typedef struct stat TTDHostStatType;
+
+#define TTDHostPathSeparator ((TTDHostCharType*)"/")
+#define TTDHostPathSeparatorChar ((TTDHostCharType)'/')
+#define TTDHostFindInvalid nullptr
+
+#define TTDHostCharConvert(X) ((char*)X)
+#define TTDHostUtf8CharConvert(X) ((TTDHostCharType*)X)
+
+size_t TTDHostStringLength(const TTDHostCharType* str)
+{
+    return strlen(TTDHostCharConvert(str));
+}
+
+void TTDHostInitEmpty(TTDHostCharType* dst)
```

Comment:
Does TTDHostCharType has to be different between _WIN32 and non-_WIN32? If you can consolidate to the same type, you don't need 2 copies of these.